packages/consent/consent-tools/src/domain/create-wrapper.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  Categories,
+  CreateWrapper,
+  AnyAnalytics,
+  CDNSettingsIntegrations,
+  CDNSettingsRemotePlugin,
+  InitOptions,
+  CreateWrapperSettings,
+  CDNSettings,
+} from '../types'
+import { validateCategories, validateOptions } from './validation'
+import { createConsentStampingMiddleware } from './consent-stamping'
+import { pipe, pick, uniq } from '../utils'
+import { AbortLoadError, LoadContext } from './load-cancellation'
+
+export const createWrapper: CreateWrapper = (createWrapperOptions) => {
+  const {
+    shouldDisableSegment,
+    shouldDisableConsentRequirement,
+    getCategories,
+    shouldLoad,
+    integrationCategoryMappings,
+    shouldEnableIntegration,
+  } = createWrapperOptions
+
+  validateOptions(createWrapperOptions)
+
+  return (analytics) => {
+    const ogLoad = analytics.load
+
+    const loadWithConsent: AnyAnalytics['load'] = async (
+      settings,
+      options
+    ): Promise<void> => {
+      // do not load anything -- segment included
+      if (await shouldDisableSegment?.()) {
+        return
+      }
+
+      const consentRequirementDisabled =
+        await shouldDisableConsentRequirement?.()
+      if (consentRequirementDisabled) {
+        // ignore consent -- just call analytics.load as usual
+        return ogLoad.call(analytics, settings, options)
+      }
+
+      // use these categories to disable/enable the appropriate device mode plugins
+      let initialCategories: Categories
+      try {
+        initialCategories =
+          (await shouldLoad?.(new LoadContext())) || (await getCategories())
+      } catch (e: unknown) {
+        // consumer can call ctx.cancel({ loadSegmentNormally: true })
+        // to load Segment but disable consent requirement
+        if (e instanceof AbortLoadError) {
+          if (e.loadSegmentNormally === true) {
+            ogLoad.call(analytics, settings, options)
+          }
+        } else {
+          console.error(e)
```

Comment:
Kind of curious on why we'd log versus throw the error (since we're in a browser context)